app/src/main/java/org/p2p/wallet/history/ui/history/HistoryPresenter.kt

```diff
@@ -70,29 +65,52 @@ class HistoryPresenter(
 
     override fun loadHistory() {
         if (transactions.isNotEmpty()) {
-            transactions = transactions
+            view?.showHistory(transactions)
             return
         }
+
         launch {
             view?.showPagingState(PagingState.InitialLoading)
             fetchHistory()
         }
     }
 
     private suspend fun fetchHistory(isRefresh: Boolean = false) {
```

Comment:
I propose to split refresh and initial loading functions

Reply:
i agree with you but it was critical issue =( not refactoring but i agree